application/views/adif/import.php

```diff
@@ -285,6 +294,35 @@
                     </form>
                 </div>
             </div>
+            <div class="tab-pane <?php if ($showtab == 'cbr') {
+                                            echo 'active';
+                                        } else {
+                                            echo 'fade';
+                                        } ?>" id="cbr" role="tabpanel" aria-labelledby="home-tab">
+                    <?php if (isset($error) && $showtab == 'cbr') { ?>
+                        <div class="alert alert-danger" role="alert">
+                            <?php echo $error; ?>
+                        </div>
+                    <?php } ?>
+
+                    <p class="card-text"><span class="badge text-bg-info">Information</span> <?= __("If you imported an ADIF file of a contest, provided by another logging software, sometimes, depending on that software, your exchanges will not be imported properly from that softwares ADIF. If you like to correct that, you can provide the Cabrillo file that this software also provides to rewrite that data in Wavelog.") ?></p>
+                    <p class="card-text"><span class="badge text-bg-warning">Important</span> <?= __("Please use this function before changing anything about the QSOs in Wavelog, as this function uses the Contest ID, as well as date and time information from both your already imported ADIF file, as well as the CBR file you are about to upload to match the QSOs and only correct relevant data.") ?></p>
+                    <form class="form" action="<?php echo site_url('cabrillo/cbrimport'); ?>" method="post" enctype="multipart/form-data">
+
+                        <div class="mb-3 row">
+                            <div class="col-md-10">
+                                <div class="form-check-inline">
+                                    <input class="form-check-input" type="checkbox" name="serial_number_present" value="1" id="serial_number_present" unchecked>
+                                    <label class="form-check-label" for="serial_number_present"><?= __("A Serial Number is part of the Exchange of this contest.") ?></label>
+                                </div>
+                                <div class="small form-text text-muted"><?= __("If unchecked, this will erase the default serial number that (for example) N1MM+ produces. If checked, it will correct the serial number if necessary.") ?></div>
```

Comment:
This is a bit misleading. While testing I did the mistake and checked that even it was wrong to check it. 

I tested with the HELVETIA Contest. In this contest I SEND an exchange but I can recv serialnumbers by non-HB9 stations. Maybe you can describe this a little bit different. As far I understood this this only means serial number I sent or am I wrong?

Reply:
You are right, this is not 100% correct.
This flag switches the behaviour to place the serial number into COL_STX / COL_SRX which are integer fields (as they should be).
So for "either exchange or serial" kind of exchange, the checkbox should be off (because the real exchange would not be convertable to integer most times and has to be written to COL_STX_STRING / COL_SRX_STRING even though it could be numerical.
I'll change this to "A Serial Number is ALWAYS part of the Exchange of this contest." to indicate this does not apply in that case, but only in a case where both partys always exchange numbers.